### PR TITLE
Terraform: 0.12 compatibility

### PR DESCRIPTION
**Pull request by chrised**
This PR makes the module compatible with Terraform 0.12, and includes the changes from #1 (GitHub Enterprise Support).

Post merging this, the module will no longer be compatible with Terraform 0.11.

### CODE REVIEW

File: weave_flux_bootstrap.tf [hunk 1]
```diff
@@ -85,38 +85,39 @@ resource "kubernetes_cluster_role_binding" "flux" {
...
+      match_labels = {
         name = "flux"
       }
     }
 
     strategy {
-      type           = "Recreate"
-      rolling_update = {}
+      type = "Recreate"
+      rolling_update {
```
Comment (chrised) [inline weave_flux_bootstrap.tf]: Apparently this isn't allowed alongside `type = "Recreate"`

File: weave_flux_bootstrap.tf [hunk 2]
```diff
@@ -139,7 +140,7 @@ resource "kubernetes_deployment" "flux" {
 
           secret {
             secret_name  = "flux-git-deploy"
-            default_mode = 0400
+            default_mode = "0400"
```
Comment (chrised) [inline weave_flux_bootstrap.tf]: The `0.12upgrade` command tried to turn this into `default_mode = 256`, so I figured a string of `0400` made more sense.

Comment (chrised) [reply weave_flux_bootstrap.tf]: Turns out 256 is 0400, who knew?

File: variables.tf [hunk 3]
```diff
@@ -2,6 +2,11 @@ variable "eks_cluster_name" {
   description = "Name of the EKS cluster upon which to configure Flux"
 }
 
+variable "github_base_url" {
+  description = "This is the target GitHub base API endpoint. Providing a value is a requirement when working with GitHub Enterprise."
+  default     = null
```
Comment (aaron-trout) [inline variables.tf]: Can you default this to the internet github?

File: weave_flux_bootstrap.tf (hunk 1 above)
Comment (aaron-trout) [reply weave_flux_bootstrap.tf]: Yup API doc agrees, probably get rid of the `rolling_update{}` block then.

Screenshot 2020-01-29 at 14 32 36

File: variables.tf (hunk 3 above)
Comment (chrised) [reply variables.tf]: `null` is the default value for the github provider, so it *should* be the internet GitHub. Will double-check though.

Comment (chrised) [reply variables.tf]: Yeah, I set one up with a base_url of `null` and it tried to talk to GitHub proper.

**Pull request merged**